format/udf-spec.md

```diff
@@ -0,0 +1,221 @@
+|-------------|------------------------------|------------------------------------------------------------------|
+| *required*  | `function-uuid`              | A UUID that identifies the function, generated once at creation. |
+| *required*  | `format-version`             | Metadata format version (must be `1`).                           |
+| *required*  | `definition`                 | List of function overloads.                                      |
+| *required*  | `definition-versions`        | List of versioned function definitions.                          |
+| *required*  | `current-definition-version` | Identifier of the current definition version.                    |
+| *optional*  | `location`                   | The storage location of metadata files.                          | 
+| *optional*  | `properties`                 | Arbitrary key-value pairs.                                       | 
+| *optional*  | `secure`                     | Whether it is a secure function. Default: `false`.               |
+| *optional*  | `doc`                        | Documentation string.                                            |
+
+### Overload
+
+Function overloads allow multiple implementations of the same function name with different signatures. Each overload has
+the following fields:
+
+| Requirement | Field name      | Description                                                                                                                                                                                        |
+|-------------|-----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| *required*  | `overload-uuid` | A UUID that identifies this function overload.                                                                                                                                                     |
+| *required*  | `parameters`    | Ordered list of function parameter definitions (name, type, optional doc). The order of parameters in this list **must exactly match** the order of arguments provided when invoking the function. |
+| *required*  | `return-type`   | Return type (scalar or struct). Example: `"string"` or `"struct<...>"`                                                                                                                             |
+| *required*  | `versions`      | List of overload versions.                                                                                                                                                                         |
+| *optional*  | `udtf`          | Whether this overload returns a table, default to `false`. If `true`, the `return-type` must be a `struct` describing the output schema.                                                           | 
```

Comment:
Reminder: We discussed converting this field to a function field type, which will include scalar, aggregate, and table functions.